website/docs/pipelines/pipelines-with-tekton.mdx

```diff
@@ -0,0 +1,312 @@
+  "reason": "Promote",
+  "reportingController": "pipeline-controller",
+  "reportingInstance": "chart-pipeline-controller-8549867565-7822g"
+}
+```
+
+In out tasks, we are using only the `involvedObject.name`,
+`involvedObject.namespace` and `message` fields:
+
+```yaml
+---
+apiVersion: triggers.tekton.dev/v1beta1
+kind: TriggerBinding
+metadata:
+  name: ww-pipeline-binding
+  namespace: ww-pipeline
+spec:
+  params:
+  - name: namespace
+    value: $(body.involvedObject.namespace)
+  - name: name
+    value: $(body.involvedObject.name)
+  - name: message
+    value: $(body.message)
+```
+
+### Tekton TriggerTemplate
+
+The template has the same parameters as the `Pipeline` resources has:
+
+```yaml
+---
+apiVersion: triggers.tekton.dev/v1beta1
+kind: TriggerTemplate
+metadata:
+  name: ww-pipeline-template
+  namespace: ww-pipeline
+spec:
+  params:
+  - name: namespace
+    default: "Unknown"
+  - name: name
+    default: "Unknown"
+  - name: message
+    default: "no message"
+  resourcetemplates:
+  - apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+      generateName: hello-goodbye-run-
+    spec:
+      pipelineRef:
+        name: hello-goodbye
+      params:
+      - name: name
+        value: $(tt.params.name)
+      - name: namespace
+        value: $(tt.params.namespace)
+      - name: message
+        value: $(tt.params.message)
+```
+
+### Tekton EventListener
+
+To access all [required resources](https://tekton.dev/docs/getting-started/triggers/#create-an-eventlistener), we need an extra service account:
+
+```yaml
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: tekton-ww-pipeline-robot
+  namespace: ww-pipeline
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: triggers-example-eventlistener-binding
+  namespace: ww-pipeline
+subjects:
+- kind: ServiceAccount
+  name: tekton-ww-pipeline-robot
+  namespace: ww-pipeline
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tekton-triggers-eventlistener-roles
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: triggers-example-eventlistener-clusterbinding
+subjects:
+- kind: ServiceAccount
+  name: tekton-ww-pipeline-robot
+  namespace: ww-pipeline
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tekton-triggers-eventlistener-clusterroles
+```
+
+With this `ServiceAccount`, we can create the `EventListener` using the
+`TriggerBinding` and `TriggerTemplate`:
+
+```yaml
+---
+apiVersion: triggers.tekton.dev/v1beta1
+kind: EventListener
+metadata:
+  name: ww-pipeline-listener
+  namespace: ww-pipeline
+spec:
+  serviceAccountName: tekton-ww-pipeline-robot
+  triggers:
+    - name: ww-pipeline-trigger
+      bindings:
+      - ref: ww-pipeline-binding
+      template:
+        ref: ww-pipeline-template
+```
+
+At this point, we should have a `Service` for our `EventListener`.
```

Comment:
Not necessarily. Yes, but no. It's enough if they add it to the cluster in any other ways if they have a hybrid solution (even with `kubectl apply`), and to be fair, if they add them all together it still works.

It's most like "see it creates a server and you can inspect it" at any point, for example if they want to `port-forward` it for testing.

I do not expect any guides here to state "and commit it to your giptops repo otherwise it will not be there".